lib/jekyll/regenerator.rb

```diff
@@ -104,7 +104,10 @@ def modified?(path)
     def add_dependency(path, dependency)
       return if (metadata[path].nil? || @disabled)
 
-      metadata[path]["deps"] << dependency unless metadata[path]["deps"].include? dependency
+      if !metadata[path]["deps"].include? dependency
+        metadata[path]["deps"] << dependency
+        add(dependency) unless metadata.include?(dependency)
```

Comment:
Well, it *should* be, but it's not a bug or anything. So, yes, we should change it, but maybe as a separate refactor/documentation issue/PR instead?